### PR TITLE
Fix 28465: Restore list of values to globalCompositeOperation page

### LINKED ISSUE #28465: globalCompositeOperation lacks list of operations
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/globalCompositeOperation

### What specific section or headline is this issue about?

Operations, which disappeared.

### What information was incorrect, unhelpful, or incomplete?

There used to be a list of all operations on this page which was very illustrative. Because a) it lists the possible values and b) it demonstrates them. Now it just says it's a "string", which is unhelpful. Which string values exactly?

### What did you expect to see?

this https://web.archive.org/web/20230201054214/https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/globalCompositeOperation

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/api/canvasrenderingcontext2d/globalcompositeoperation`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/globalCompositeOperation
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/canvasrenderingcontext2d/globalcompositeoperation/index.md
* Last commit: https://github.com/mdn/content/commit/bda6da832e5682d6247e1c20790e7b0aa0ec45e1
* Document last modified: 2023-08-03T20:27:59.000Z

</details>

### PR DESCRIPTION
**Pull request by wbamberg**
Fixes https://github.com/mdn/content/issues/28465.

Fixed as suggested in https://github.com/mdn/content/issues/28465#issuecomment-1681604111.

### CODE REVIEW

File: files/en-us/web/api/canvasrenderingcontext2d/globalcompositeoperation/index.md [hunk 1]
```diff
@@ -17,7 +17,60 @@ See also [Compositing and clipping](/en-US/docs/Web/API/Canvas_API/Tutorial/Comp
...
+- `"darken"`
+  - : Retains the darkest pixels of both layers.
+- `"lighten"`
+  - : Retains the lightest pixels of both layers.
+- `"color-dodge"`
+  - : Divides the bottom layer by the inverted top layer.
+- `"color-burn"`
+  - : Divides the inverted bottom layer by the top layer, and then inverts the result.
+- `"hard-light"`
+  - : A combination of multiply and screen like overlay, but with top and bottom layer swapped.
```
Comment (sideshowbarker) [inline files/en-us/web/api/canvasrenderingcontext2d/globalcompositeoperation/index.md:59]:
```suggestion
  - : A combination of `multiply` and `screen`-like overlay, but with top and bottom layer swapped.
```

File: files/en-us/web/api/canvasrenderingcontext2d/globalcompositeoperation/index.md [hunk 2]
```diff
@@ -17,7 +17,60 @@ See also [Compositing and clipping](/en-US/docs/Web/API/Canvas_API/Tutorial/Comp
...
+- `"copy"`
+  - : Only the new shape is shown.
+- `"xor"`
+  - : Shapes are made transparent where both overlap and drawn normal everywhere else.
+- `"multiply"`
+  - : The pixels of the top layer are multiplied with the corresponding pixel of the bottom layer. A darker picture is the result.
+- `"screen"`
+  - : The pixels are inverted, multiplied, and inverted again. A lighter picture is the result (opposite of multiply)
+- `"overlay"`
+  - : A combination of multiply and screen. Dark parts on the base layer become darker, and light parts become lighter.
```
Comment (sideshowbarker) [inline files/en-us/web/api/canvasrenderingcontext2d/globalcompositeoperation/index.md:49]:
```suggestion
  - : A combination of `multiply` and `screen`. Dark parts on the base layer become darker, and light parts become lighter.
```

File: files/en-us/web/api/canvasrenderingcontext2d/globalcompositeoperation/index.md [hunk 3]
```diff
@@ -17,7 +17,60 @@ See also [Compositing and clipping](/en-US/docs/Web/API/Canvas_API/Tutorial/Comp
...
+- `"lighten"`
+  - : Retains the lightest pixels of both layers.
+- `"color-dodge"`
+  - : Divides the bottom layer by the inverted top layer.
+- `"color-burn"`
+  - : Divides the inverted bottom layer by the top layer, and then inverts the result.
+- `"hard-light"`
+  - : A combination of multiply and screen like overlay, but with top and bottom layer swapped.
+- `"soft-light"`
+  - : A softer version of hard-light. Pure black or white does not result in pure black or white.
```
Comment (sideshowbarker) [inline files/en-us/web/api/canvasrenderingcontext2d/globalcompositeoperation/index.md:61]:
```suggestion
  - : A softer version of `hard-light`. Pure black or white does not result in pure black or white.
```

File: files/en-us/web/api/canvasrenderingcontext2d/globalcompositeoperation/index.md [hunk 4]
```diff
@@ -17,7 +17,60 @@ See also [Compositing and clipping](/en-US/docs/Web/API/Canvas_API/Tutorial/Comp
...
+- `"difference"`
+  - : Subtracts the bottom layer from the top layer or the other way round to always get a positive value.
```
Comment (sideshowbarker) [inline files/en-us/web/api/canvasrenderingcontext2d/globalcompositeoperation/index.md:63]:
```suggestion
  - : Subtracts the bottom layer from the top layer — or the other way round — to always get a positive value.
```

File: files/en-us/web/api/canvasrenderingcontext2d/globalcompositeoperation/index.md [hunk 5]
```diff
@@ -17,7 +17,60 @@ See also [Compositing and clipping](/en-US/docs/Web/API/Canvas_API/Tutorial/Comp
...
+- `"exclusion"`
+  - : Like difference, but with lower contrast.
```
Comment (sideshowbarker) [inline files/en-us/web/api/canvasrenderingcontext2d/globalcompositeoperation/index.md:65]:
```suggestion
  - : Like `difference`, but with lower contrast.
```

File: files/en-us/web/api/canvasrenderingcontext2d/globalcompositeoperation/index.md [hunk 6]
```diff
@@ -51,6 +104,320 @@ ctx.fillRect(50, 50, 100, 100);
...
+  canvas2.width = width;
+  canvas2.height = height;
+  lightMix();
+  colorSphere();
+  runComposite();
+  return;
+};
+```
+
+And this code, runComposite(), handles the bulk of the work, relying on a number of utility functions to do the hard parts.
```
Comment (sideshowbarker) [inline files/en-us/web/api/canvasrenderingcontext2d/globalcompositeoperation/index.md:200]:
```suggestion
And this code, `runComposite()`, handles the bulk of the work, relying on a number of utility functions to do the hard parts.
```

File: files/en-us/web/api/canvasrenderingcontext2d/globalcompositeoperation/index.md [hunk 7]
```diff
@@ -17,7 +17,60 @@ See also [Compositing and clipping](/en-US/docs/Web/API/Canvas_API/Tutorial/Comp
...
+- `"destination-over"`
+  - : New shapes are drawn behind the existing canvas content.
+- `"destination-in"`
+  - : The existing canvas content is kept where both the new shape and existing canvas content overlap. Everything else is made transparent.
+- `"destination-out"`
+  - : The existing content is kept where it doesn't overlap the new shape.
+- `"destination-atop"`
+  - : The existing canvas is only kept where it overlaps the new shape. The new shape is drawn behind the canvas content.
+- `"lighter"`
+  - : Where both shapes overlap the color is determined by adding color values.
```
Comment (sideshowbarker) [inline files/en-us/web/api/canvasrenderingcontext2d/globalcompositeoperation/index.md:39]:
```suggestion
  - : Where both shapes overlap, the color is determined by adding color values.
```

File: files/en-us/web/api/canvasrenderingcontext2d/globalcompositeoperation/index.md [hunk 8]
```diff
@@ -17,7 +17,60 @@ See also [Compositing and clipping](/en-US/docs/Web/API/Canvas_API/Tutorial/Comp
...
+- `"copy"`
+  - : Only the new shape is shown.
+- `"xor"`
+  - : Shapes are made transparent where both overlap and drawn normal everywhere else.
+- `"multiply"`
+  - : The pixels of the top layer are multiplied with the corresponding pixel of the bottom layer. A darker picture is the result.
```
Comment (sideshowbarker) [inline files/en-us/web/api/canvasrenderingcontext2d/globalcompositeoperation/index.md:45]:
```suggestion
  - : The pixels of the top layer are multiplied with the corresponding pixels of the bottom layer. A darker picture is the result.
```

File: files/en-us/web/api/canvasrenderingcontext2d/globalcompositeoperation/index.md [hunk 9]
```diff
@@ -17,7 +17,60 @@ See also [Compositing and clipping](/en-US/docs/Web/API/Canvas_API/Tutorial/Comp
...
+- `"screen"`
+  - : The pixels are inverted, multiplied, and inverted again. A lighter picture is the result (opposite of multiply)
```
Comment (sideshowbarker) [inline files/en-us/web/api/canvasrenderingcontext2d/globalcompositeoperation/index.md:47]:
```suggestion
  - : The pixels are inverted, multiplied, and inverted again. A lighter picture is the result (opposite of `multiply`)
```

**Review by sideshowbarker — APPROVED** [review]

**Comment by wbamberg** [comment]
Thanks for the review @sideshowbarker !

**Pull request merged**